src/vs/workbench/api/browser/mainThreadTask.ts

```diff
@@ -649,7 +649,7 @@ export class MainThreadTask implements MainThreadTaskShape {
 						return;
 					}
 				}
-				reject(new Error('Task to terminate not found'));
+				reject(new Error(`Task to terminate not found for ID: ${id}, tasks: ${tasks.map(t => `ID: ${t._id} Label: ${t._label}`)}`));
```

Comment:
I've learned we typically don't want to add customizable text like this as then a lot of distinct errors will end up showing in telemetry.

Reply:
> should this just be a warning instead of an error or is this condition unexpected?

This feels like a warning that should either silently fail (as the request is done already), or show a notification. If we just want a notification and that's happening because of this `Error` bubbling up, changing it to `ErrorNoTelemetry` will keep that.